api/v1/ytsaurus_types.go

```diff
@@ -285,6 +285,10 @@ type InstanceSpec struct {
 	// Component config for native RPC bus transport.
 	//+optional
 	NativeTransport *RPCTransportSpec `json:"nativeTransport,omitempty"`
+	// DNSConfig allows customizing the DNS settings for the pods.
+	//+optional
+	DNSConfig *corev1.PodDNSConfig `json:"dnsConfig,omitempty"`
+	DNSPolicy *corev1.DNSPolicy    `json:"dnsPolicy,omitempty"`
```

Comment:
Since corev1.DNSPolicy is string under the hood and "" value is default zero value in the pod spec I suggest we just use `corev1.DNSPolicy` as type, not pointer, so we don't have to handle 2 empty values.